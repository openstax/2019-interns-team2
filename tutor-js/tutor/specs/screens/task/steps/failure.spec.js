import Failure from '../../../../src/screens/task/step/failure';
import Raven from '../../../../src/models/app/raven';
jest.mock('../../../../src/models/app/raven');


describe('Task Failure', () => {
  let props;

  beforeEach(() => {
    props = {
      task: {
        api: {
          errors: {},
        },
      },
      step: {
        id: 1234,
        api: {
          errors: {},
        },
      },
    };
  });

  it('matches snapshot', () => {
    expect(<Failure {...props} />).toMatchSnapshot();
  });

  it('logs step message', () => {
    const fail = mount(<Failure {...props} />);
    expect(Raven.log).toHaveBeenCalledWith(
      expect.stringContaining('Failed to load assignment step id: 1234')
    );
    fail.unmount();
  });

  it('logs task message', () => {
    props.task.api.hasErrors = true;
    props.task.api.errors = { foo: 'bar' };
    const fail = mount(<Failure {...props} />);
    expect(Raven.log).toHaveBeenCalledWith(
      expect.stringContaining('Failed to load assignment, errors: Foo bar')
    );
    fail.unmount();
  });

});
