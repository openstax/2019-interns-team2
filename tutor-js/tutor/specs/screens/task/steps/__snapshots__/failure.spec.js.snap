// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Failure matches snapshot 1`] = `
<Failure
  step={
    Object {
      "api": Object {
        "errors": Object {},
      },
      "id": 1234,
    }
  }
  task={
    Object {
      "api": Object {
        "errors": Object {},
      },
    }
  }
/>
`;
